### PR TITLE
Move CUDAWrapper::MatrixFree tests to new folder

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

**Comment by Rombur** [comment]
Can you also move the `cuda_evaluate` tests from `cuda` to `matrix_free_kokkos`

File: tests/matrix_free_kokkos/cuda_evaluate_1d_shape.cc [hunk 1]
```diff
@@ -32,7 +32,6 @@
 
 #include "../tests.h"
 
-namespace CUDA = LinearAlgebra::CUDAWrappers;
```
Comment (peterrum) [inline tests/matrix_free_kokkos/cuda_evaluate_1d_shape.cc:35]: Modification 1.

File: tests/matrix_free_kokkos/cuda_evaluate_1d_shape.cc [hunk 2]
```diff
@@ -196,7 +195,6 @@ main()
   deallog.attach(logfile);
 
   Kokkos::initialize();
-  init_cuda();
```
Comment (peterrum) [inline tests/matrix_free_kokkos/cuda_evaluate_1d_shape.cc:199]: Modification 2.

**Review by peterrum — COMMENTED** [review]
@Rombur I have moved the files. However, had to do minor modifications. Could you check these?

**Review by Rombur — APPROVED** [review]
Thanks.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**